tests/src/core/testqgsexpression.cpp

```diff
@@ -1011,6 +1013,9 @@ class TestQgsExpression: public QObject
       QTest::newRow( "flip_coordinates point" ) << "geom_to_wkt(flip_coordinates(geom_from_wkt('POINT(1 2)')))" << false << QVariant( "Point (2 1)" );
 
       // string functions
+      QChar separator = QLocale().groupSeparator();
```

Comment:
It's safe to hardcode a ',' here - the tests all run under the c locale. And I'd be concerned that if the groupSeperator was '' then this test won't actually test what it's supposed to.

Can you also add tests for >>999,999 (i.e. with multiple separators)

Reply:
Also a test for a number with > 3 decimals (your regex doesn't work here :cry: ), .e.g 1000.000001 -> 1,000.000,001

Reply:
Oh darn -- any idea on a better regex string then?